### ISSUE TITLE
Add "Undo" button to Layer Properties

### ISSUE THREAD

**Issue by qgib** [qgis/QGIS#23109]
Author Name: **Dan Jacobson** (@jidanni)
Original Redmine Issue: [15170](https://issues.qgis.org/issues/15170)

Redmine category:gui

---
On Layer Properties there are four buttons at the bottom:

OK, Cancel, Apply, and Help (which does not show any help, by the way.)

Well you forgot Undo.

Without Undo, if there is any mishap, which is extremely likely, the
user must quit without saving changes, and start qgis all over again.

You might think that closing Layer Properties, and clicking Edit > Undo
would help. But I am sorry to inform you that that item is greyed out.
Furthermore, CTRL+Z does nothing.

**Comment by alexbruy** [comment]
@jidanni does using Layer Styling panel solves your issue?

**Comment by jidanni** [comment]
> @jidanni does using Layer Styling panel solves your issue?

I don't know, because I haven't used Qgis in years.

**Comment by DelazJ** [comment]
@gioman I'm not sure we should close this feature request. This is a valid need IMO: I just fall in the trap of doing changes in the layer properties dialog, applying them and then realize that the previous state was better. I had to configure again (hard to remember the exact previous settings) while an Undo button would have made it straightforward.
The Layer Styling is far from covering all the tabs provided by the layer properties.

**Comment by DelazJ** [comment]
I don't think this request should go stale. Having a way to undo changes in the layer properties (and not only for the tabs supported by the Layer Styling panel) makes sense imho

**Comment by DelazJ** [comment]
can we keep this open? Seems a valid request to me.

**Comment by AlisterH** [comment]
Also see previous discussion on this thread, including standards for 3 platforms which all agree that the "cancel" behaviour is wrong https://github.com/qgis/QGIS/issues/14075#issuecomment-495612508
;)
